src/ints/int10_modes.cpp

```diff
@@ -1999,56 +2179,54 @@ static cga_colors_t configure_cga_colors()
 	        control->GetSection("render"));
 	assert(render_section);
 
-	const auto cga_colors_prefs = split(render_section->Get_string("cga_colors"));
+	const std::string cga_colors_prefs = render_section->Get_string("cga_colors");
 
-	if (cga_colors_prefs.size() == 0) {
+	if (cga_colors_prefs.empty()) {
 		LOG_WARNING("INT10: No value specified for 'cga_colors', using default CGA colors");
 		return cga_colors_default;
 	}
 
-	const auto preset = cga_colors_prefs[0];
-
-	if (preset == "default")
+	if (cga_colors_prefs == "default")
 		return cga_colors_default;
 
-	else if (preset == "tandy")
+	else if (starts_with("tandy", cga_colors_prefs))
 		return handle_cga_colors_prefs_tandy(cga_colors_prefs);
 
-	else if (preset == "ibm5153")
+	else if (starts_with("ibm5153", cga_colors_prefs))
 		return handle_cga_colors_prefs_ibm5153(cga_colors_prefs);
 
-	else if (preset == "tandy-warm")
+	else if (cga_colors_prefs == "tandy-warm")
 		return cga_colors_tandy_warm;
 
-	else if (preset == "agi-amiga-v1")
+	else if (cga_colors_prefs == "agi-amiga-v1")
 		return cga_colors_agi_amiga_v1;
 
-	else if (preset == "agi-amiga-v2")
+	else if (cga_colors_prefs == "agi-amiga-v2")
 		return cga_colors_agi_amiga_v2;
 
-	else if (preset == "agi-amiga-v3")
+	else if (cga_colors_prefs == "agi-amiga-v3")
 		return cga_colors_agi_amiga_v3;
 
-	else if (preset == "agi-amigaish")
+	else if (cga_colors_prefs == "agi-amigaish")
 		return cga_colors_agi_amigaish;
 
-	else if (preset == "scumm-amiga")
+	else if (cga_colors_prefs == "scumm-amiga")
 		return cga_colors_scumm_amiga;
 
-	else if (preset == "colodore")
+	else if (cga_colors_prefs == "colodore")
 		return cga_colors_colodore_sat50;
 
-	else if (preset == "colodore-sat")
+	else if (cga_colors_prefs == "colodore-sat")
 		return cga_colors_colodore_sat60;
 
 	else {
-		if (cga_colors_prefs.size() == 1) {
-			LOG_WARNING("INT10: Invalid 'cga_colors' preset name '%s', "
-			            "using default CGA colors",
-			            preset.c_str());
+		auto maybe_cga_colors = parse_cga_colors(cga_colors_prefs);
+		if (maybe_cga_colors.has_value()) {
+			return maybe_cga_colors.value();
+		} else {
+			LOG_WARNING("INT10: Using default CGA colors");
```

Comment:
Here's a similar combo `has_value` and `value` that PVS is flagging:

![2022-08-14_10-05_1](https://user-images.githubusercontent.com/1557255/184548900-4854b994-7a12-4424-8ef0-7491fd1b91df.png)

Looks like the `value_or(...)` function can return a default if the value doesn't exist.

Maybe we can use it with something like, to take care of the PVS issue:

```c++
const auto cga_colors = parse_cga_colors(cga_colors_prefs);
if (!cga_colors) {
	LOG_WARNING("INT10: Using default CGA colors");
}
return cga_colors.value_or(cga_colors_default);
```

Reply:
Much better and nice touch with the use of `or` 👍🏻

Reply:
I'm glad you found this 'optional' smart pointer; I hadn't heard of it before. 

It's really nice seeing C++ no longer being afraid of being more expressive with the base language, with member functions like: ".value_or(default)", which take care of a big percentage of what users are trying to do.. (in prior years, I think these committees were too up tight about including richer handling and function names like this 😂)

Reply:
Yeah, these additions are nice for sure. What I'm missing a lot is a proper "either" implementation that is quite useful for functional style error handling. For example, from the token parser I could have returned something like a `std::either<std::string, RGBEntry>`, where the left side would be the the error message, or the right side the actual result.